src/settingsview.cpp

```diff
@@ -102,23 +119,21 @@ void SettingsView::browseMonitorDir()
     if (dir == monitorDir || dir.isEmpty()) {
         return;
     }
-    if (confirmDirDialog(dir, gt("monitor-dir-dialog-msg"), gt("monitor-dir-dialog-title"))) {
+    if (confirmDialogMonitorDir(dir, false)) {
         KiwixApp::instance()->getSettingsManager()->setMonitorDir(dir);
         ui->monitorClear->show();
         KiwixApp::instance()->getLibrary()->loadMonitorDir(dir);
         KiwixApp::instance()->getWatcher()->addPath(dir);
-    } else {
-        QMessageBox msgBox;
-        msgBox.setText(gt("monitor-directory-invalid"));
-        msgBox.exec();
     }
 }
 
 void SettingsView::clearMonitorDir()
 {
     auto dir = QString();
-    if (confirmDirDialog(dir, gt("monitor-clear-dir-dialog-msg"), gt("monitor-clear-dir-dialog-title"))) {
+    if (confirmDialogMonitorDir(dir, true)) {
```

Comment:
Since in this case you don't need to replace the `{{DIRECTORY}}` parameter in the message you can call `confirmDirDialog()` here directly:

```
if (confirmDirDialog(gt("monitor-clear-dir-dialog-msg"), gt("monitor-clear-dir-dialog-title"))) {
```